packages/eslint-plugin/test/import-relative-within-subpackage.spec.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+const mockModule = require('../utils/mockModule');
+const mock = mockModule('../utils/import-aliases');
+mock.getAllSpinnakerPackages.mockImplementation(() => ['core', 'amazon', 'kubernetes']);
+
+const path = require('path');
+const ruleTester = require('../utils/ruleTester');
+const rule = require('../rules/import-relative-within-subpackage');
+
+ruleTester.run('import-relative-within-subpackage', rule, {
+  valid: [
+    {
+      filename: '/root/spinnaker/deck/app/scripts/modules/amazon/package/amazon_source_file.ts',
+      code: `import { Anything } from '../subpackage/foo';`,
+    },
+    {
+      filename: '/root/spinnaker/deck/app/scripts/modules/amazon/src/package/amazon_source_file.ts',
+      code: `import { Anything } from '../subpackage/foo';`,
+    },
+    {
+      filename: '/root/spinnaker/deck/app/scripts/modules/core/subpackage/nest/core_source_file.ts',
+      code: `import { Anything } from 'core/subpackage';`,
+    },
+    {
+      // kubernetes subpackages are nested under v1 or v2 directory so treat them differently
```

Comment:
Not for long! https://github.com/spinnaker/governance/blob/master/rfc/eol_kubernetes_v1.md